django/contrib/admin/options.py

```diff
@@ -309,6 +309,19 @@ def get_readonly_fields(self, request, obj=None):
         """
         return self.readonly_fields
 
+    def get_uneditable_fields(self, request, obj=None):
```

Comment:
Hi All,
I was a little surprised to see a get_uneditable_fields method in addition to get_readonly_fields. I suppose that provides the best backward compatibility, though it was a little confusing on first glance to have _two_ methods with very similar names.
I don't really have a better suggestion though. _Maybe_ a get_field_names() that handles the possibility of declared_fieldsets, and a can_edit() that checks add and change permissions. It makes the code more complicated, but it might be a little more intuitive. Just a thought.